src/pattern-library/components/patterns/LayoutPatterns.js

```diff
@@ -0,0 +1,132 @@
+import { useState } from 'preact/hooks';
+
+import { LabeledButton } from '../../../';
+
+import {
+  PatternPage,
+  Pattern,
+  PatternExamples,
+  PatternExample,
+} from '../PatternPage';
+
+export default function LayoutPatterns() {
+  const [showExample1, setShowExample1] = useState(false);
+  const [showExample2, setShowExample2] = useState(false);
+  const [showExample3, setShowExample3] = useState(false);
+  return (
+    <PatternPage title="Layout">
+      <Pattern title="Fixed-Centered Positioning">
+        <p>
+          The <code>fixed-centered</code> layout pattern centers an element both
+          horizontally and vertically within the entire viewport.
+        </p>
+        <PatternExamples>
+          <PatternExample details="Centering an element vertically and horizontally within the viewport">
+            <div>
+              <LabeledButton
+                variant="primary"
+                onClick={() => setShowExample1(true)}
+              >
+                Show example
+              </LabeledButton>
+              <div
+                className="hyp-u-fixed-centered"
+                style={`width:450px;${
+                  showExample1 ? 'visibility:visible' : 'visibility:hidden'
+                }`}
+              >
+                <div className="hyp-card">
```

Comment:
Maybe these two divs could be combined into one:

```suggestion
              <div
                className="hyp-u-fixed-centered hyp-card"
                style={`width:450px; ${
                  showExample1 ? 'visibility:visible' : 'visibility:hidden'
                }`}
              >
```

Reply:
Maybe also for the ones below: `className="hyp-u-overlay hyp-u-fixed-centered"` or `className=hyp-u-overlay hyp-u-fixed-centered hyp-card"`

Reply:
I'd like to keep the two separate div elements because I want to prevent any case in which styles applied by (in this case) `hyp-card` are interfering with `hyp-u-fixed-centered`. `hyp-u-fixed-centered` is intended to be a container, while the `div.hyp-card` is the "thing inside the container."

This will mirror how this pattern will be used in the `Modal` component, which I expect will be something like:

```jsx
<>
  <div className="Modal__overlay" />
  <div className="Modal__container" />
  <div className="Dialog" />
</>
```

Where `Modal__overlay` inherits from `overlay` and `Modal__container` from `fixed-centered` and `Dialog` is inside the `fixed-centered` "container."